tests/components/switchbot/conftest.py

```diff
@@ -1,22 +1,203 @@
+) -> None:
+    """Set up the Switchbot service."""
+    await async_setup_component(hass, "bluetooth", {})
+    inject_bluetooth_service_info(hass, bluetooth_service_info)
+
+
+@pytest.fixture
+def mock_config_entry(
+    switchbot_model: SwitchbotModel,
+    encrypted_data: dict[str, Any],
+    bluetooth_service_info: BluetoothServiceInfoBleak,
+    setup_service: None,
+    switchbot_device: None,
+) -> MockConfigEntry:
+    """Fixture to create a MockConfigEntry for a Switchbot device."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_ADDRESS: bluetooth_service_info.address,
+            CONF_NAME: "test-name",
+            CONF_SENSOR_TYPE: SUPPORTED_MODEL_TYPES[switchbot_model],
+        }
+        | encrypted_data,
+        unique_id="aabbccddeeff",
+    )
+
+
+@pytest.fixture
+def encrypted_data(switchbot_model: SwitchbotModel) -> dict[str, Any]:
+    """Fixture to create encrypted data for Switchbot devices."""
+    if switchbot_model in ENCRYPTED_MODELS:
+        return {
+            CONF_KEY_ID: "ff",
+            CONF_ENCRYPTION_KEY: "ffffffffffffffffffffffffffffffff",
+        }
+    return {}
+
+
+@pytest.fixture
+async def switchbot_device(
+    mock_switchbot_blind_tilt: AsyncMock,
+    mock_switchbot_roller_shade: AsyncMock,
+    mock_switchbot_light_strip: AsyncMock,
+    mock_switchbot_fan: AsyncMock,
+    mock_switchbot_vacuum: AsyncMock,
+) -> None:
+    """Fixture to create a mock Switchbot device."""
+
+
+@pytest.fixture
+async def mock_switchbot_blind_tilt() -> AsyncGenerator[AsyncMock]:
+    """Fixture to create a mock Switchbot device."""
+    with patch(
+        "homeassistant.components.switchbot.fan.switchbot.SwitchbotBlindTilt.update"
+    ):
+        yield
+
+
+@pytest.fixture
+async def mock_switchbot_roller_shade() -> AsyncGenerator[AsyncMock]:
+    """Fixture to create a mock Switchbot device."""
+    with patch(
+        "homeassistant.components.switchbot.fan.switchbot.SwitchbotRollerShade.update"
+    ):
+        yield
+
+
+@pytest.fixture
+async def mock_switchbot_light_strip() -> AsyncGenerator[AsyncMock]:
+    """Fixture to create a mock Switchbot device."""
+    with patch(
+        "homeassistant.components.switchbot.fan.switchbot.SwitchbotLightStrip.update"
+    ):
+        yield
+
+
+@pytest.fixture
+async def mock_switchbot_fan() -> AsyncGenerator[AsyncMock]:
+    """Fixture to create a mock Switchbot device."""
+    with patch("homeassistant.components.switchbot.fan.switchbot.SwitchbotFan.update"):
+        yield
+
+
+@pytest.fixture
+async def mock_switchbot_vacuum() -> AsyncGenerator[AsyncMock]:
+    """Fixture to create a mock Switchbot device."""
+    with patch(
+        "homeassistant.components.switchbot.vacuum.switchbot.SwitchbotVacuum.update"
+    ):
+        yield
+
+
+@pytest.fixture
+async def mock_switchbot_curtain() -> AsyncGenerator[dict[str, AsyncMock]]:
+    """Fixture to create a mock Switchbot device."""
```

Comment:
Something like this might work
```
    with patch.multiple(
        "homeassistant.components.switchbot.cover.switchbot.SwitchbotCurtain",
        open=patch(),
        close=patch(),
        stop=patch(),
        set_position=patch(),
    ) as mocks:
        yield mocks
```

or AyncMock() instead of patch()